### PR TITLE
Remove php 7.4 and Sylius 1.9

### PR DESCRIPTION
**Pull request by oallain**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no
| Fixed issue | 

- [x] Removed PHP 7.4 because that release is no longer supported
- [x] Removed Sylius 1.9 because that release support only PHP 7.4
- [x] Update dependencies
- [x] Update code with PHP 8.0 syntax

### CODE REVIEW

File: tests/PHPUnit/Filter/ProductFilterTest.php [hunk 1]
```diff
@@ -30,14 +30,11 @@ final class ProductFilterTest extends ApiTestCase
     /** @var ProductFiltersRules */
     private $productFiltersRules;
 
-    /** @var ProductFilter */
-    private $productFilter;
+    private ?object $productFilter = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Filter/ProductFilterTest.php:33]: TODO: fix type

File: tests/PHPUnit/Retriever/FamilyRetrieverTest.php [hunk 2]
```diff
@@ -9,17 +9,16 @@
...
 /**
  * @internal
+ *
  * @coversNothing
  */
 final class FamilyRetrieverTest extends ApiTestCase
 {
-    /** @var FamilyRetrieverInterface */
-    private $familyRetriever;
+    private ?object $familyRetriever = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Retriever/FamilyRetrieverTest.php:21]: TODO: fix type

File: tests/PHPUnit/Task/Family/AddFamilyVariationAxeTaskTest.php [hunk 3]
```diff
@@ -18,15 +17,15 @@
...
  * @coversNothing
  */
 final class AddFamilyVariationAxeTaskTest extends AbstractTaskTest
 {
     /** @var AkeneoTaskProvider */
     private $taskProvider;
 
-    /** @var EntityRepository */
-    private $productGroupRepository;
+    private ?object $productGroupRepository = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Task/Family/AddFamilyVariationAxeTaskTest.php:28]: TODO: fix type

File: tests/PHPUnit/Task/ProductModel/AddOrUpdateProductModelTaskTest.php [hunk 4]
```diff
@@ -21,18 +19,17 @@
...
  * @coversNothing
  */
 final class AddOrUpdateProductModelTaskTest extends AbstractTaskTest
 {
     /** @var AkeneoTaskProvider */
     private $taskProvider;
 
-    /** @var ProductRepositoryInterface */
-    private $productRepository;
+    private ?object $productRepository = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Task/ProductModel/AddOrUpdateProductModelTaskTest.php:30]: TODO: fix type

File: tests/PHPUnit/Task/ProductOption/CreateUpdateTaskTest.php [hunk 5]
```diff
@@ -16,12 +16,12 @@
...
 /**
  * @internal
+ *
  * @coversNothing
  */
 final class CreateUpdateTaskTest extends AbstractTaskTest
 {
-    /** @var \Synolia\SyliusAkeneoPlugin\Provider\TaskProvider */
-    private $taskProvider;
+    private ?object $taskProvider = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Task/ProductOption/CreateUpdateTaskTest.php:24]: TODO: fix type

File: tests/PHPUnit/Task/ProductOption/DeleteTaskTest.php [hunk 6]
```diff
@@ -11,12 +11,12 @@
...
 /**
  * @internal
+ *
  * @coversNothing
  */
 final class DeleteTaskTest extends AbstractTaskTest
 {
-    /** @var \Synolia\SyliusAkeneoPlugin\Provider\TaskProvider */
-    private $taskProvider;
+    private ?object $taskProvider = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Task/ProductOption/DeleteTaskTest.php:19]: TODO: fix type

File: tests/PHPUnit/Transformer/AkeneoAttributeToSyliusAttributeTransformerTest.php [hunk 7]
```diff
@@ -7,17 +7,16 @@
...
 /**
  * @internal
+ *
  * @coversNothing
  */
 final class AkeneoAttributeToSyliusAttributeTransformerTest extends AbstractKernelTestCase
 {
-    /** @var AkeneoAttributeToSyliusAttributeTransformerInterface */
-    private $akeneoAttributeToSyliusAttributeTransformer;
+    private ?object $akeneoAttributeToSyliusAttributeTransformer = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Transformer/AkeneoAttributeToSyliusAttributeTransformerTest.php:19]: TODO: fix type

File: tests/PHPUnit/TypeMatcher/Attribute/AttributeTypeMatcherTest.php [hunk 8]
```diff
@@ -21,14 +21,14 @@
...
+ *
  * @coversNothing
  */
 final class AttributeTypeMatcherTest extends AbstractKernelTestCase
 {
     private const FAKE_AKENEO_ATTRIBUTE_TYPE = 'my_fake_attribute_type';
 
-    /** @var AttributeTypeMatcher */
-    private $attributeTypeMatcher;
+    private ?object $attributeTypeMatcher = null;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/TypeMatcher/Attribute/AttributeTypeMatcherTest.php:31]: TODO: fix type

**Review by TheGrimmChester — APPROVED** [review]

File: src/Form/DataTransformer/AssetEntityDataTransformer.php [hunk 9]
```diff
@@ -12,19 +12,13 @@ final class AssetEntityDataTransformer implements DataTransformerInterface
...
+    public function transform(mixed $value): string
     {
-        $json = json_encode($value);
-
-        if (false === $json) {
-            throw new AssetTransformException('Could not transform asset array to json.');
-        }
-
-        return $json;
+        return json_encode($value, \JSON_THROW_ON_ERROR);
```
[the remark this answers is not in the document]
Comment (oallain) [reply src/Form/DataTransformer/AssetEntityDataTransformer.php]: Fixed

File: src/Form/DataTransformer/MetricDataTransformer.php [hunk 10]
```diff
@@ -22,19 +22,13 @@ public function transform($value): string
...
         }
 
-        $json = json_encode($value);
-
-        if (false === $json) {
-            throw new MetricTransformException('Could not transform metric array to json.');
-        }
-
-        return $json;
+        return json_encode($value, \JSON_THROW_ON_ERROR);
```
[the remark this answers is not in the document]
Comment (oallain) [reply src/Form/DataTransformer/MetricDataTransformer.php]: Fixed

File: src/Form/DataTransformer/ReferenceEntityDataTransformer.php [hunk 11]
```diff
@@ -5,23 +5,16 @@
...
+    public function transform(mixed $value): string
     {
-        $json = json_encode($value);
-
-        if (false === $json) {
-            throw new ReferenceEntityTransformException('Could not transform reference entity array to json.');
-        }
-
-        return $json;
+        return json_encode($value, \JSON_THROW_ON_ERROR);
```
[the remark this answers is not in the document]
Comment (oallain) [reply src/Form/DataTransformer/ReferenceEntityDataTransformer.php]: Fixed

**Pull request merged**